cocos/ui/UIListView.cpp

```diff
@@ -540,12 +540,38 @@ void ListView::selectedItemEvent(TouchEventType event)
                 (_listViewEventListener->*_listViewEventSelector)(this, LISTVIEW_ONSELECTEDITEM_END);
             }
             if (_eventCallback) {
-                _eventCallback(this, EventType::ON_SELECTED_ITEM_END);
+                _eventCallback(this, EventType::ON_SELECTED_ITEM_END, pTouch);
             }
             if (_ccEventCallback)
             {
                 _ccEventCallback(this, static_cast<int>(EventType::ON_SELECTED_ITEM_END));
             }
+            //end calc
+            if (event == TouchEventType::ENDED)
+            {
+                //only ended will know last choose one，begin just highlight
+                Widget* parent = sender;
+                while (parent)
+                {
+                    if (parent && (parent->getParent() == _innerContainer))
```

Comment:
`parent` is necessarily not null here. The test should be `if (parent->getParent() == _innerContainer)`.